src/precess_cd.jl

```diff
@@ -0,0 +1,78 @@
+# This file is a part of AstroLib.jl. License is MIT "Expat".
+
+function _precess_cd{T<:AbstractFloat}(cd::Matrix{T}, epoch1::T, epoch2::T, crval_old::Vector{T},
+                                       crval_new::Vector{T}, FK4::Bool)
+    crvalold = deg2rad.(crval_old)
+    crvalnew = deg2rad.(crval_new)
+    t = deg2rad((epoch2 - epoch1)*0.001)
+
+    if FK4
+        st = (epoch1 - 1900)*0.001
+        c = t*(20046.85 - st*(85.33 + st*0.37) + t*(-42.67 - st*0.37 - t*41.8))/3600
+    else
+        st = (epoch1 - 2000)*0.001
+        c = t*(20043.109 - st*(85.33 + st*0.217) + t*(-42.665 - st*0.217 - t*41.8))/3600
+    end
+    pole_ra = 0
+    pole_dec = 90
+
+    if (epoch1 == 2000 && epoch2 == 1950) || (epoch1 == 1950 && epoch2 == 2000)
+        pole_ra, pole_dec = bprecess(pole_ra, pole_dec)
+    else
+        pole_ra, pole_dec = precess(pole_ra, pole_dec, epoch1, epoch2, FK4=FK4)
+    end
+    sind1, sind2 = sin.([crvalold[2], crvalnew[2]])
+    cosd1, cosd2 = cos.([crvalold[2], crvalnew[2]])
+    sinra = sin(crvalnew[1] - deg2rad.(pole_ra))
+    cosfi = (cos(c) - sind1*sind2)/(cosd1*cosd2)
+    sinfi = (abs(sin(c))* sinra)/cosd1
+    r = [[cosfi -sinfi]; [sinfi cosfi]]
+    cd = cd * r
```

Comment:
Uhm, you're right!  I was doing tests and noted that the input `cd` matrix was changing, but probably I had replaced the line with `cd .= cd * r`, which does mutate.